enhancements/installer/infrastructure-apis-custom-trust.md

```diff
@@ -0,0 +1,236 @@
+---
+title: infrastructure-apis-custom-trust
+authors:
+  - "@abhinavdahiya"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-06-30
+last-updated: 2020-06-30
+status: implementable
+see-also:
+  - "/enhancements/proxy/global-cluster-egress-proxy.md"  
+---
+
+# Custom Trust for Infrastructure APIs
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The users should be allowed to provide a custom trust bundle for the purpose of communicating with infrastructure APIs, which should be available to all the operators so that they can configure their SDKs. The operators will be able to pull the trust bundle from a defined ConfigMap or use the [`config.openshift.io/inject-trusted-cabundle`][egress-proxy-proposal] mechanism to mount a volume with trust already configured. The later is important because a majority of our operators communicate with infrastructure APIs and configuring the SDKs with custom trust is tedious and error prone similar to the PROXY setup.
+
+
+## Motivation
+
+OpenShift components interact quite a bit with the underlying infrastructure using APIs to provide it's users various features and a majority of our operators today interact with infrastructure APIs. For our commercial platforms like AWS, Azure, GCP's public cloud trusting the secured API endpoints is not a problem, but for other platforms like vSphere, OpenStack or even private clouds from AWS or Azure we need our operators to trust user-provided bundles to securely communicate to the APIs.
+
+### Goals
+
+1. Allow users to provide custom trust for APIs using openshift-installer.
```

Comment:
Installer already has the `.additionalTrustBundle` in install-config.yaml , so I do not think we need something separate.

Also almost all the clouds only depend on environment for credentials and nothing else. openstack depending on environment for configuration like the custom trust is anti-pattern to me.

Reply:
> This pem file ends up in the `openshift-config/cloud-provider-config` configmap under the `cacert.pem` key.

I like this pattern since the cloud.yaml  file already requires this information if the kube control-plane is going to be able to communicate with the cloud provider.  I'm ok with an operator taking a different, separate configuration to build the cloud.yaml if that is preferred, but defining trust for the cloud provider in the cloud provider configmap seems like a  very good idea.